### DISCUSSION TITLE
知识库引用内容应该放在systen提示词下

### DISCUSSION THREAD

**Discussion by gaord** [labring/FastGPT#1121] [General]
目前知识库引用 是放在human下面。我觉得放在system提示词下面比较好，可以减少对对话生成的干扰。放在human下面有时候模型回答会受知识库引用内容的干扰。

**Comment by c121914yu** [comment]
放system太容易被忽略了

**Reply by Zhenyi-Wang** [reply]
ChatGPT是可以的，可能和模型有关，希望能够设置。否则多语言场景下经常导致AI一直用中文回复

**Reply by gaord** [reply]
建议改变这种情况。模型忽略系统提示是模型的问题。现在国内模型也在进步。
尝试了通过工作流机制改变，发现这个流程机制太硬了，需要的设置无法配置出来。

**Comment by gaord** [comment]
建议增加一个设置，具体放在哪里流程图上设定

**Comment by Zhenyi-Wang** [comment]
最近遇到知识库泄露的情况，选择的是“知识库+对话引导”模式。这个模式下，不论用户说什么，都会触发知识库搜索，然后拼接用户问题，有时候会有歧义。比如：
```
// 情况1：用户有感而发，回复一个6

FastGPT会把知识库插入成：
<Data>
1. 问题：xxx
回答：xxx

2. 问题：xxx
回答：xxx

……
</Data>

问题："""6"""

然后AI会直接把知识库的第6条回复出来

// 情况2：用户先发了一条中文，AI回复，然后用户补充“翻译成英文”

<Data>
一些知识点
一些知识点
一些知识点
……
</Data>

问题："""翻译成英文"""

这时AI会把整个<Data>内的知识翻译成英文。
```

后来尝试自定义工作流，把知识库作为一个工具给AI调用，似乎可以解决，只是要在提示词上花些工夫，避免有时候该查的时候不去查。
